seatunnel-connectors/seatunnel-connector-spark-clickhouse/src/main/scala/org/apache/seatunnel/spark/sink/Clickhouse.scala

```diff
@@ -131,12 +198,77 @@ class Clickhouse extends SparkBatchSink {
     schema
   }
 
+  private def getClusterShardList(conn: ClickHouseConnectionImpl, clusterName: String,
+                                  database: String, port: String): ListBuffer[Shard] = {
+    val rs = conn.createStatement().executeQuery(
+      s"select shard_num,shard_weight,replica_num,host_name,host_address," +
+        s"port from system.clusters where cluster = '$clusterName'")
+    // TODO The port will be used for data communication of the tcp protocol in the future
+    // port is tcp protocol, need http protocol port at now
+    val nodeList = mutable.ListBuffer[Shard]()
+    while (rs.next()) {
+      nodeList += new Shard(rs.getInt(1), rs.getInt(2),
+        rs.getInt(3), rs.getString(4), rs.getString(5),
+        port, database)
```

Comment:
@BenJFan It would be better to get it by field here

Reply:
Use field is ok, but use offset could be faster.

Reply:
This code need change? If yes I will do that.